src/telliot_feeds/sources/evm_call.py

```diff
@@ -104,7 +125,7 @@ def get_response(self) -> Optional[Any]:
 
         return (result, ts)
 
-    async def fetch_new_datapoint(self) -> OptionalDataPoint[Any]:
+    async def fetch_new_datapoint(self, block_number: BlockIdentifier = "latest") -> OptionalDataPoint[Any]:
```

Comment:
I agree although i do not intend to use it for the reporter, rather I intend to use it only for the DVM's `general_fetch_new_datapoint()`. For the reporter, the block_number should always be the default `latest` so it will match the dataspec. But let me know if you think I should still test it out :)